assets/js/base/hooks/cart/use-store-cart.js

```diff
@@ -113,14 +113,17 @@ export const useStoreCart = ( options = { shouldSelect: true } ) => {
 			);
 			const shippingRatesLoading = store.isCustomerDataUpdating();
 			const { receiveCart } = dispatch( storeKey );
-			const billingAddress = decodeAddress( cartData.billingAddress );
+			const billingAddress = decodeValues( cartData.billingAddress );
 			const shippingAddress = cartData.needsShipping
-				? decodeAddress( cartData.shippingAddress )
+				? decodeValues( cartData.shippingAddress )
 				: billingAddress;
+			const cartFees = Array.isArray( cartData.fees )
+				? cartData.fees.map( ( fee ) => decodeValues( fee ) )
+				: [];
```

Comment:
Don't think this isArray check is necessary, as `fees` will always be coming back as an array. For example it is `[]` when no fees are available.

Reply:
Hmmm, I'm not sure about this one. Cart data comes directly from the server, so there is the danger that the data we receive is not in the form we expect it, so I think it's good to have this kind of protections? At least, we have the same for `cartItems`, `cartItemErrors`, `extensions` and `shippingRates`:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/f880fdbe9cfb00c38854a5d2647b64fab435699c/assets/js/base/hooks/cart/use-store-cart.js#L122-L135

However, I also noticed that removing this protection caused several tests to fail. In this case, I think the issue was with tests and I fixed them here 6247633bd973e15168059e96a04762cb125de6c9.

Let me know if you agree with the resolution @ralucaStan.